tests/test_connections/test_ledger/test_ledger.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021-2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the tests of the ledger connection module."""
+
+
+import asyncio
+from asyncio import Task
+from typing import Any, List
+from unittest import mock
+
+import pytest
+from aea.configurations.base import ConnectionConfig
+from aea.connections.base import ConnectionStates
+
+from packages.valory.connections.ledger.connection import LedgerConnection
+
+
+def dummy_task_wrapper(waiting_time: float, result: Any) -> Task:
+    """Create a dummy task that simply waits for a given `waiting_time` and returns a given `result`."""
+
+    async def dummy_task(*_: Any) -> None:
+        """Wait for the given `waiting_time` and return the given `result`."""
+        await asyncio.sleep(waiting_time, result)
+
+    task = asyncio.create_task(dummy_task())
+    return task
+
+
+async def dummy_receive(receiving_tasks: List[asyncio.Future]) -> None:
+    """A dummy receive method that simply waits for the first task to complete and returns the done and pending."""
+    await asyncio.wait(receiving_tasks, return_when=asyncio.FIRST_COMPLETED)
+
+
+class TestLedgerConnection:
```

Comment:
Haven't found the issue yet, but I suggest it would be better in any case to write a custom RequestDispatcher which implements `dummy_task` as a "performative" and is triggered via the message flow. That gets us closer to the real failure.

Reply:
Managed to recreate the issue in a more minimal way.